File: app/backend/dto/payment.py
from typing import Optional

from ninja import Schema


class CreatePaymentDTO(Schema):
    tg_user_id: str
    payment_id: str
    сount_generations: int
    count_video_generations: Optional[int]
    amount: str
    learn_model: bool
    is_first_payment: bool
    promo: bool
    count_generations_for_gift: int
    count_generations_video_for_gift: Optional[int]

    
class PaymentDTO(Schema):
    tg_user_id: str
    payment_id: str
    status: bool
    сount_generations: int
    amount: str
    promo: bool
    count_generations_for_gift: int
    count_generations_video_for_gift: Optional[int]

File: app/stat/internal/entity/payment.go
package entity

import "time"

type Payment struct {
	TgUserID              string    `gorm:"size:30;column:tg_user_id"`                // TG User ID
	PaymentID             string    `gorm:"size:200;uniqueIndex;column:payment_id"`   // Payment ID (уникальный)
	Status                bool      `gorm:"default:false;column:status"`              // Статус платежа
	CountGenerations      uint      `gorm:"default:0;column:count_generations"`       // Кол-во генераций
	CountVideoGenerations uint      `gorm:"default:0;column:count_video_generations"` // Кол-во генераций видео
	Amount                string    `gorm:"size:20;column:amount"`                    // Сумма платежа
	LearnModel            bool      `gorm:"default:false;column:learn_model"`         // Обучение модели
	IsFirstPayment        bool      `gorm:"default:false;column:is_first_payment"`    // Первый платеж
	CreatedAt             time.Time `gorm:"autoCreateTime;column:created_at"`         // Дата создания
}
